sdk/schemaregistry/Azure.Data.SchemaRegistry/samples/Sample02_Serialization.md

```diff
@@ -0,0 +1,227 @@
+# Using the `SchemaRegistrySerializer`
+
+The following shows how to use the `SchemaRegistrySerializer` to serialize and deserialize JSON payloads containing Schema Registry schema identifiers and serialized data. The default serialization is JSON, but the serializer is customizable to any type of serialization.
+
+## Table of contents
+- [Implementing `SchemaValidator`](#implementing-schemavalidator)
+- [Serialize and deserialize JSON data using the Event Hub EventData model](#serialize-and-deserialize-json-data-using-the-event-hub-eventdata-model)
+- [Serialize and deserialize JSON data using `MessageContent` directly](#serialize-and-deserialize-json-data-using-messagecontent-directly)
+- [Pluggable object serialization](#pluggable-object-serialization)
+- [Troubleshooting](#troubleshooting)
+- [Next steps](#next-steps)
+- [Contributing](#contributing)
+
+## Implementing `SchemaValidator`
+
+The `SchemaValidator` is an abstract class that must be implemented and passed into the `SchemaRegistryJsonSerializer` constructor. This allows you to choose the third-party JSON Schema package you would like to use to generate schemas from .NET types and validate .NET objects against JSON schemas. Implementing both `SchemaValidator.Validate` and `SchemaValidator.GenerateSchema` is required.
+
+The following is an outline of how an implemented `SchemaValidator` may look:
+```C# Snippet:SampleSchemaRegistryJsonSchemaGeneratorImplementation
+internal class SampleJsonValidator : SchemaValidator
+{
+    public override void Validate(Object data, Type dataType, string schemaDefinition)
+    {
+        // Your implementation using the third-party library of your choice goes here. This method throws
+        // an exception if the data argument is not valid according to the schemaDefinition.
+
+        List<Exception> validationErrors = SampleValidationClass.SampleValidationMethod(schemaDefinition, data, dataType);
+
+        if (validationErrors.Count  > 0)
+        {
+            throw new AggregateException(validationErrors);
+        }
+
+        return;
+    }
+
+    public override string GenerateSchema(Type dataType)
+    {
+        // Your implementation using the third-party library of your choice goes here.
+
+        return "<< SCHEMA GENERATED FROM DATATYPE PARAMETER >>";
+    }
+}
+```
+
+The following is a sample implementation using `Newtonsoft.Json.Schema`:
```

Comment:
I have mixed feelings on calling out `Newtonsoft.Json.Schema` here, even if we're using its syntax for illustration.   On one hand, it's good to point folks at an implementation, and Newtonsoft is a well-known entity.  On the other hand, its a commercial product and I'm not sure that we can even legally do an unofficial endorsement.

Reply:
Yeah I thought about that as well, my idea was that including a sample implementation seems like it would be helpful for customers. I'll take it out, we could potentially add a fully fake implementation, or just omit this piece.